### ISSUE TITLE
No error for implementing abstract methods

### ISSUE THREAD

**Issue by shreydesai** [python/mypy#1889]
Java, among other statically typed languages, provides errors for those trying to create a body for a method that is abstract:

<img width="352" alt="screen shot 2016-07-17 at 10 54 10 am" src="https://cloud.githubusercontent.com/assets/12023280/16902215/004fb7bc-4c0d-11e6-8426-0589df487546.png">

The following code doesn't give an error. I'm not sure if this is an architectural design by mypy; to some, it doesn't make sense to create a body for a method that needs to be overridden.

```python
from abc import ABCMeta, abstractmethod

class A(metaclass=ABCMeta):
    @abstractmethod
    def f(self) -> None:
        print('Hello world')
```

**Comment by kirbyfan64** [comment]
[In the docs for `abc`](https://docs.python.org/3/library/abc.html):

> Unlike Java abstract methods, these abstract methods may have an implementation. This implementation can be called via the [`super()`](https://docs.python.org/3/library/functions.html#super) mechanism from the class that overrides it. This could be useful as an end-point for a super-call in a framework that uses cooperative multiple-inheritance.

So this is definitely intentional.
